test/ProductConstructionService.ScenarioTests/EndToEndFlowLogic.cs

```diff
@@ -8,17 +8,10 @@
 
 namespace ProductConstructionService.ScenarioTests;
 
-internal class EndToEndFlowLogic : ScenarioTestBase
+internal partial class ScenarioTestBase
```

Comment:
I agree this class was weird but maybe it'd be better if we made it an abstract child of `ScenarioTestBase` and inherit from it only where we need it? Seems like these are just 2 classes.

Reply:
Oh, I see, it was like that but also we did `new()` it...

Reply:
Done